src/devices/bus/a2bus/a2superdrive.cpp

```diff
@@ -0,0 +1,349 @@
+ */
+uint8_t a2bus_superdrive_device::read_c800(uint16_t offset)
+{
+	unsigned address;
+
+	if (offset < 0x400)
+		address = (m_bank_select << 10) + offset;
+	else
+		address = 0x7c00 + offset - 0x400;
+
+	return m_ram[address];
+}
+
+void a2bus_superdrive_device::write_c800(uint16_t offset, uint8_t data)
+{
+	unsigned address;
+
+	if (offset < 0x400)
+		address = (m_bank_select << 10) + offset;
+	else
+		address = 0x7c00 + offset - 0x400;
+
+	m_ram[address] = data;
+}
+
+
+/* uc 65c02 i/o at $0a00 */
+void a2bus_superdrive_device::m65c02_w(offs_t offset, uint8_t value)
+{
+	// $00-$0f = swim registers
+	// $40 = head sel low
+	// $41 = head sel high
+	// $80 = diagnostic led on
+	// $81 = diagnostic led off
+
+	floppy_image_device *floppy = nullptr;
+
+	if (offset < 16)
+	{
+		m_fdc->write(offset, value);
+		return;
+	}
+
+	switch (offset)
+	{
+		case 0x40:
+			m_side = 0;
+			floppy = m_fdc->get_floppy();
+			if (floppy) floppy->ss_w(m_side);
+			break;
+
+		case 0x41:
+			m_side = 1;
+			floppy = m_fdc->get_floppy();
+			if (floppy) floppy->ss_w(m_side);
+			break;
+
+		case 0x80:
+			logerror("LED on");
+			break;
+
+		case 0x81:
+			logerror("LED off");
+			break;
+
+		default:
+			logerror("write($0a%02x,%02x)\n", offset, value);
+			break;
+	}
+}
+
+uint8_t a2bus_superdrive_device::m65c02_r(offs_t offset)
+{
+	floppy_image_device *floppy = nullptr;
+
+	if (offset < 16)
+		return m_fdc->read(offset);
+
+	switch (offset)
+	{
+		case 0x40:
+			if (machine().side_effects_disabled()) break;
+			m_side = 0;
+			floppy = m_fdc->get_floppy();
+			if (floppy) floppy->ss_w(m_side);
+			break;
+
+		case 0x41:
+			if (machine().side_effects_disabled()) break;
+			m_side = 1;
+			floppy = m_fdc->get_floppy();
+			if (floppy) floppy->ss_w(m_side);
+			break;
+
+		case 0x80:
+			logerror("LED on");
+			break;
+
+		case 0x81:
+			logerror("LED off");
+			break;
+
+		default:
+			logerror("read($0a%02x)\n", offset);
+			break;
+	}
+	return 0;
+}
+
+
+void a2bus_superdrive_device::devsel_w(uint8_t devsel)
+{
+	floppy_image_device *floppy = nullptr;
+
+	switch (devsel)
+	{
+		case 1:
+			floppy = m_fdc->subdevice<floppy_connector>("0")->get_device();
+			break;
+		case 2:
+			floppy = m_fdc->subdevice<floppy_connector>("1")->get_device();
+			break;
+	}
+
+	m_fdc->set_floppy(floppy);
+}
+
+void a2bus_superdrive_device::phases_w(uint8_t phases)
+{
+	floppy_image_device *floppy = m_fdc->get_floppy();
+	if (floppy)
+		floppy->seek_phase_w(phases);
+}
+
+
+void a2bus_superdrive_device::sel35_w(int sel35)
+{
+}
+
+void a2bus_superdrive_device::hdsel_w(int hdsel)
+{
+	/* Q3/HDSEL pin (ISM MODE register bit 5) is used to control the clock speed */
+	/* MFM runs at 15.6672, GCR at 15.6672/2 */
+
+	m_fdc->set_clock_scale( hdsel ? 1.0 : 0.5);
+}
+
+} // anonymous namespace
+
+
+DEFINE_DEVICE_TYPE_PRIVATE(A2BUS_SUPERDRIVE, device_a2bus_card_interface, a2bus_superdrive_device, "superdrive", "Apple II 3.5\" Disk Controller Card")
```

Comment:
I didn’t notice it last time, but the “short names” for Apple II bus cards all have an `a2` prefix as a kind of crude namespacing.  Remember device short names all live in a single namespace shared with driver short names.